SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialWorkerFlags.cpp

```diff
@@ -22,20 +22,56 @@ void USpatialWorkerFlags::ApplyWorkerFlagUpdate(const Worker_FlagUpdateOp& Op)
 		FString NewValue = FString(UTF8_TO_TCHAR(Op.value));
 		FString& ValueFlag = WorkerFlags.FindOrAdd(NewName);
 		ValueFlag = NewValue;
-		OnWorkerFlagsUpdated.Broadcast(NewName, NewValue);
+		OnAnyWorkerFlagUpdated.Broadcast(NewName, NewValue);
+		if (FOnWorkerFlagUpdated* OnWorkerFlagUpdated = OnWorkerFlagUpdatedMap.Find(NewName))
+		{
+			OnWorkerFlagUpdated->Broadcast(NewValue);
+		}
 	}
 	else
 	{
 		WorkerFlags.Remove(NewName);
 	}
 }
 
-void USpatialWorkerFlags::BindToOnWorkerFlagsUpdated(const FOnWorkerFlagsUpdatedBP& InDelegate)
+void USpatialWorkerFlags::RegisterAndInvokeFlagUpdatedCallback(const FString& InFlagName, const FOnWorkerFlagUpdatedBP& InDelegate)
 {
-	OnWorkerFlagsUpdated.Add(InDelegate);
+	RegisterFlagUpdatedCallback(InFlagName, InDelegate);
+	if (FString* ValueFlag = WorkerFlags.Find(InFlagName))
+	{
+		InDelegate.Execute(*ValueFlag);
+	}
+}
+
+void USpatialWorkerFlags::RegisterFlagUpdatedCallback(const FString& InFlagName, const FOnWorkerFlagUpdatedBP& InDelegate)
+{
+	FOnWorkerFlagUpdated& OnWorkerFlagUpdated = OnWorkerFlagUpdatedMap.FindOrAdd(InFlagName);
+	OnWorkerFlagUpdated.Add(InDelegate);
+}
+
+void USpatialWorkerFlags::UnregisterFlagUpdatedCallback(const FString& InFlagName, const FOnWorkerFlagUpdatedBP& InDelegate)
+{
+	if (FOnWorkerFlagUpdated* OnWorkerFlagUpdated = OnWorkerFlagUpdatedMap.Find(InFlagName))
+	{
+		OnWorkerFlagUpdated->Remove(InDelegate);
+	}
+}
+
+void USpatialWorkerFlags::RegisterAndInvokeAnyFlagUpdatedCallback(const FOnAnyWorkerFlagUpdatedBP& InDelegate)
+{
+	RegisterAnyFlagUpdatedCallback(InDelegate);
+	for (const auto& flagPair : WorkerFlags)
+	{
+		OnAnyWorkerFlagUpdated.Broadcast(flagPair.Key, flagPair.Value);
```

Comment:
hmm on second thought this might have undesirable side effects should just execute the delegate directly here.